meteor/client/ui/Settings/components/triggeredActions/TriggeredActionsEditor.tsx

```diff
@@ -208,11 +214,11 @@ export const TriggeredActionsEditor: React.FC<IProps> = function TriggeredAction
 	)
 
 	useSubscription(PubSub.partInstances, {
-		rundownId: rundown?._id ?? false,
+		rundownId: rundown?._id ?? protectString('Ignore All'),
```

Comment:
I suspect that as we move over to React Hooks, this is a pattern that we might see more of. Maybe it should be wrapped into something like `protectedInvalidId()` or something, which would return a static protectedString?

Reply:
yeah that could work, or my suggestion of `we should discuss whether we should have these mongo selectors as parameters to the publications.` would be a proper solution.
In this particular case the publication could be done as `useSubscription(PubSub.partInstances, rundownPlaylist?.activationId, rundown?._id)`. That would let the backend easily handle this and simply not run the mongo query when the rundownId is undefined.

We could probably do something similar with the current api. If this publication gets a subscriber with no rundownId, then return an empty collection. It feels a little dangerous to assume that we will always a rundownId, so the api change would be better, but it might be safe for the current queries (I havent checked yet)

Ill try and remember to bring up this publications api topic at the next devmeet

Reply:
Or more shortterm, to preserve `the flexibility to subscribe to data in almost any shape is very handy`, perhaps we add a new `isValid: boolean` parameter to some of the publications? That way the ui can flag a query as invalid/incomplete and meteor will skip executing it